packages/flutter/test/material/stepper_test.dart

```diff
@@ -368,6 +368,105 @@ void main() {
     expect(find.text('2'), findsOneWidget);
   });
 
+  testWidgets('Stepper custom builder test', (WidgetTester tester) async {
+    final ControlsWidgetBuilder builder =
+      (BuildContext context, VoidCallback onStepContinue, VoidCallback onStepCancel) {
+        return Container(
+          margin: const EdgeInsets.only(top: 16.0),
+          child: ConstrainedBox(
+            constraints: const BoxConstraints.tightFor(height: 48.0),
+            child: Row(
+              children: <Widget>[
+                FlatButton(
+                  onPressed: onStepContinue,
+                  color: Colors.blue,
+                  textColor: Colors.white,
+                  textTheme: ButtonTextTheme.normal,
+                  child: const Text('Let us continue!'),
+                ),
+                Container(
+                  margin: const EdgeInsetsDirectional.only(start: 8.0),
+                  child: FlatButton(
+                    onPressed: onStepCancel,
+                    textColor: Colors.red,
+                    textTheme: ButtonTextTheme.normal,
+                    child: const Text('Cancel This!'),
+                  ),
+                ),
+              ],
+            ),
+          ),
+        );
+      };
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Center(
+          child: Material(
+            child: Stepper(
+              controlsBuilder: builder,
+              steps: const <Step>[
+                Step(
+                  title: Text('A'),
+                  state: StepState.complete,
+                  content: SizedBox(
+                    width: 100.0,
+                    height: 100.0,
+                  ),
+                ),
+                Step(
+                  title: Text('B'),
+                  content: SizedBox(
+                    width: 100.0,
+                    height: 100.0,
+                  ),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // 2 because stepper creates a set of controls for each step
+    expect(find.text('Let us continue!'), findsNWidgets(2));
+    expect(find.text('Cancel This!'), findsNWidgets(2));
+  });
+
+  testWidgets('Stepper custom builder test 2', (WidgetTester tester) async {
```

Comment:
Not sure what this is testing?

Reply:
Just testing to make sure that if we don't provide custom controls, everything still builds as expected. But maybe this is covered in other tests?